src/metabase/models/dashboard_card.clj

```diff
@@ -345,6 +345,14 @@
              dashcards))
       dashcards)))
 
+(defn dashcard-comparator
+  "Comparator that determines which of two dashcards comes first in the layout order used for pulses.
+  This is the same order used on the frontend for the mobile layout. Orders cards left-to-right, then top-to-bottom"
+  [dashcard-1 dashcard-2]
+  (if-not (= (:row dashcard-1) (:row dashcard-2))
+    (compare (:row dashcard-1) (:row dashcard-2))
+    (compare (:col dashcard-1) (:col dashcard-2))))
```

Comment:
```suggestion
  [{row-1 :row col-1 :col} {row-2 :row col-2 :col}]
  (if (= row-1 row-2)
    (compare col-1 col-2)
    (compare row-1 row-2)
```

^there are several distinct suggestions here, each of which are a little silly, feel free to pick and choose as you see fit. Another one is I think `(if (not= ...` reads better than `(if-not (= ...`.

Reply:
(Ah, I now see you didn't even write the original code in the first place.)